completion.go

```diff
@@ -8,43 +8,80 @@ import (
 )
 
 const (
-	completionCommandName = "generate-completion"
-	completionFlagName    = "generate-shell-completion"
-	completionFlag        = "--" + completionFlagName
+	completionCommandName = "completion"
+
+	// This flag is supposed to only be used by the completion script itself, to generate completions on the fly.
+	completionFlagName = "generate-shell-completion"
+	completionFlag     = "--" + completionFlagName
 )
 
 var (
 	//go:embed autocomplete
 	autoCompleteFS embed.FS
 
-	shellCompletions = map[string]renderCompletion{
+	shellCompletions = map[string]renderCompletionFunc{
 		"bash": getCompletion("autocomplete/bash_autocomplete"),
 		"ps":   getCompletion("autocomplete/powershell_autocomplete.ps1"),
-		"zsh":  getCompletion("autocomplete/zsh_autocomplete"),
-		"fish": func(c *Command) (string, error) {
+		"zsh": func(cmd *Command, appName string) (string, error) {
+			return genZshCompletion(appName), nil
+		},
+		"fish": func(c *Command, appName string) (string, error) {
 			return c.ToFishCompletion()
 		},
 	}
 )
 
-type renderCompletion func(*Command) (string, error)
+type renderCompletionFunc func(cmd *Command, appName string) (string, error)
 
-func getCompletion(s string) renderCompletion {
-	return func(c *Command) (string, error) {
+func getCompletion(s string) renderCompletionFunc {
+	return func(cmd *Command, appName string) (string, error) {
 		b, err := autoCompleteFS.ReadFile(s)
 		return string(b), err
 	}
 }
 
-func buildCompletionCommand() *Command {
+func genZshCompletion(appName string) string {
+	return fmt.Sprintf(`#compdef %[1]s
```

Comment:
you could still load this content from autocompleteFS as before and then do a search/replace on a template string with appName. That way if other shells need the same logic we can reuse.

Reply:
Thanks, I prefer to keep this inside of the Go code though. imho makes it easier to see what is passed inside